api/python/provisioner/commands/check.py

```diff
@@ -249,38 +326,47 @@ def to_dict(self) -> dict:
         return result
 
 
-class DecisionMaker(ABC):
+class DecisionMaker:
+
+    """ Basic Decision Maker class. """
+
+    @staticmethod
+    def format_checks(*check_entries: CheckEntry) -> str:
+        """
+        Format all passed check entries to a single string
+
+        :param CheckEntry check_entries: one or more CheckEntry objects
 
-    """Abstract Decision Maker class"""
+        :return: string representation of all passed check entries
+        """
+        return "; ".join(str(check) for check in check_entries)
 
-    @abstractmethod
     def make_decision(self, check_result: CheckResult):
         """
         Make a decision based on given check_result
 
         :return:
         """
+        if check_result.is_failed:
+            # Get non-critical errors first
+            warnings = check_result.get_checks(critical=False)
```

Comment:
@dmitrydb  The same concern here as for SWUpdate before: do we need all non critical checks here or we need all failed ones?

Reply:
Here we need to fetch all non-critical failed checks. I fixed that. 
I left only the `get_checks` method with different parameters to specify the returning set of checks by characteristic like critical, failed.